### ISSUE TITLE
Deleted subscription is retryable

### ISSUE THREAD

**Issue by olayinkasf** [dehora/nakadi-java#226]
After deleting a subscription, the stream processor still retries and onError is never called
```

2017-07-06 10:34:37.631  INFO 14557 --- [akadi-java-io-0] NakadiClient                             : retryable_exception_consumer class nakadi.NotFoundException Not Found; Subscription with id "126d6459-bf7c-4af9-bb8b-0e695232fe6d" does not exist (404)

nakadi.NotFoundException: Not Found; Subscription with id "126d6459-bf7c-4af9-bb8b-0e695232fe6d" does not exist (404)
	at nakadi.ProblemSupport.throwProblem(ProblemSupport.java:41) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.throwProblem(OkHttpResource.java:316) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.handleError(OkHttpResource.java:312) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.throwIfError(OkHttpResource.java:297) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.requestThrowingInner(OkHttpResource.java:158) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.requestThrowingInner(OkHttpResource.java:167) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.lambda$requestThrowing$2(OkHttpResource.java:96) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.internal.operators.observable.ObservableDefer.subscribeActual(ObservableDefer.java:32) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.Observable.subscribe(Observable.java:10842) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.Observable.blockingFirst(Observable.java:4727) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.OkHttpResource.requestThrowing(OkHttpResource.java:96) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.StreamProcessorRequestFactory.requestStreamConnection(StreamProcessorRequestFactory.java:41) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.StreamProcessorRequestFactory.onCall(StreamProcessorRequestFactory.java:33) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.StreamProcessorRequestFactory.lambda$createCallable$0(StreamProcessorRequestFactory.java:21) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.internal.operators.flowable.FlowableUsing.subscribeActual(FlowableUsing.java:48) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.Flowable.subscribe(Flowable.java:13013) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.Flowable.subscribe(Flowable.java:12960) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.internal.operators.flowable.FlowableSubscribeOn$SubscribeOnSubscriber.run(FlowableSubscribeOn.java:82) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:260) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at nakadi.shadow.io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:225) ~[nakadi-java-client-0.9.0.jar:0.9.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[na:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[na:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745) ~[na:1.8.0_121]

```

**Comment by dehora** [comment]
😀 I thought we might see things like this after the requests to make the consumer default retryable instead of retry on certain exceptions. Have a couple of questions

- Just to confirm, is this 0.9.x?

- What are you trying to do here? I'm wondering why a stream processor is connecting to a deleted subscription.

- What do you expect  to happen?

**Comment by olayinkasf** [comment]
- It's on version 0.9.0

- I was trying to recreate a bug we found in our usage.

- When the processor is running, I deleted the subscription outside the service and I expected the processor to stop retrying and shutdown since the subscription doesn't exist anymore.

**Comment by dehora** [comment]
> I expected the processor to stop retrying and shutdown since the subscription doesn't exist anymore.

👍  shifting to feature.

**Comment by dehora** [comment]
@satybald the #229 patch is available in [0.9.1](https://bintray.com/dehora/maven/nakadi-java-client/0.9.1) - thanks!

**Comment by satybald** [comment]
tested it locally, works as expected.
